src/components/comparison/comparison-side/comparison-side.tsx

```diff
@@ -393,101 +456,130 @@ export const ComparisonSide = ({
+      : [];
+    const activeLayersInSelectedGroup = activeLayers.filter((activeLayer) =>
+      activeLayerIdsFromRoot.includes(activeLayer.id)
+    );
 
-            const newLayers = [...childrenToSave, selectedExample];
+    return [...activeLayersInSelectedGroup, layer];
+  };
 
-            if (parentLayer) {
-              newLayers.push(parentLayer);
-            }
+  const getActiveLayerIds = (
+    layers: LayerExample[],
+    activeLayerIds: string[] = []
+  ) => {
+    for (const layer of layers) {
+      activeLayerIds.push(layer.id);
 
-            setLayers(newLayers);
-            changedLayers = newLayers;
-          }
-          break;
-        }
+      if (layer?.layers?.length) {
+        getActiveLayerIds(layer?.layers, activeLayerIds);
       }
-
-      setPreventTransitions(false);
-      onChangeLayers && changedLayers.length && onChangeLayers(changedLayers);
     }
+
+    return activeLayerIds;
   };
 
-  const getSelectedExampleById = (
-    id?: string
-  ): { selectedExample: LayerExample | null; type: LayerType } => {
-    let selectedExample: LayerExample | null = null;
-    let type: LayerType = LayerType.single;
+  const onLayerSelectHandler = (
+    layer: LayerExample,
+    rootLayer?: LayerExample
+  ) => {
+    const isGroup = !!layer?.layers?.length;
+    const isUnitLayer = !rootLayer && !isGroup;
+    const isLeaf = rootLayer && !isGroup;
+    const isMainGroup = isGroup && !rootLayer;
 
-    for (const example of examples) {
-      if (example.id === id) {
-        selectedExample = example;
+    let newActiveLayers: LayerExample[] = [];
 
-        if (example.children?.length) {
-          type = LayerType.parent;
-        }
+    switch (true) {
+      case isUnitLayer:
+        newActiveLayers = [layer];
+        break;
+      case isGroup:
+        newActiveLayers = handleSelectGroupLayer(layer, isMainGroup, rootLayer);
+        break;
+      case isLeaf:
+        newActiveLayers = handleSelectLeafLayer(layer, rootLayer);
         break;
-      }
-
-      for (const childExample of example.children || []) {
-        if (childExample.id === id) {
-          selectedExample = childExample;
-          type = LayerType.child;
-          break;
-        }
-      }
     }
 
-    return { selectedExample, type };
+    setActiveLayers(newActiveLayers);
+    setPreventTransitions(false);
+    const activeLayersIds = getActiveLayerIds(newActiveLayers);
+    onChangeLayers &&
+      newActiveLayers.length &&
+      onChangeLayers(examples, activeLayersIds);
   };
 
   const onLayerDeleteHandler = (id: string) => {
     const idsToDelete = [id];
 
-    const layerToDelete = layers.find((layer) => layer.id === id);
-    const childIds = layerToDelete?.children?.map((child) => child.id) || [];
+    const layerToDelete = activeLayers.find((layer) => layer.id === id);
+    const childIds = layerToDelete
+      ? flattenLayersTree(layerToDelete).map((layer) => layer.id)
```

Comment:
```suggestion
      ? getAllLayersInGroupsTree(layerToDelete).map((layer) => layer.id)
```
Can we reuse this one? It is used to form active layers. Using it to delete look more consistent.

Reply:
DOne